examples/ClusterStabilityTest/Messages/Protos_protoactor.cs

```diff
@@ -0,0 +1,152 @@
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Google.Protobuf;
+using Proto;
+using Proto.Cluster;
+using Proto.Remote;
+
+namespace Messages
+{
+    public static class Grains
+    {
+        internal static Func<IHelloGrain> _HelloGrainFactory;
+
+        public static void HelloGrainFactory(Func<IHelloGrain> factory) 
+        {
+            _HelloGrainFactory = factory;
+            Remote.RegisterKnownKind("HelloGrain", Actor.FromProducer(() => new HelloGrainActor()));
+        } 
+
+        public static HelloGrainClient HelloGrain(string id) => new HelloGrainClient(id);
+    }
+
+    public interface IHelloGrain
+    {
+        Task<HelloResponse> SayHello(HelloRequest request);
+    }
+
+    public class HelloGrainClient
+    {
+        private readonly string _id;
+
+        public HelloGrainClient(string id)
+        {
+            _id = id;
+        }
+
+        public Task<HelloResponse> SayHello(HelloRequest request) => SayHello(request, CancellationToken.None);
+
+        public async Task<HelloResponse> SayHello(HelloRequest request, CancellationToken ct)
+        {
+            
+            var gr = new GrainRequest
+            {
+                MethodIndex = 0,
+                MessageData = request.ToByteString()
+            };
+
+            async Task<HelloResponse> Inner() 
+            {
+retry:
+                //resolve the grain
+                var (pid, statusCode) = await Cluster.GetAsync(_id, "HelloGrain", ct);
+                if (statusCode == ResponseStatusCode.Unavailable)
+                {
+                    await Task.Delay(100);
+                    goto retry;
```

Comment:
Hack:
In the code generated grains, there was no handling of the unavailable status, this causes the client to fail with unavailable exception.

Adding this hacky retry loop solves this.
So some form of retry for x time should definitely be added to the code generator, both c# and go

Reply:
I see that we return err in Go:
https://github.com/AsynkronIT/protoactor-go/blob/dev/examples/cluster/shared/protos_protoactor.go#L50

Which might be fine there, but throwing in C# forcing users to try catch to retry feels too heavy.

I think that in both C# and Go, it would be nice to have some form of retry x times mechanism in place to avoid confusion.

e.g. it is likely that a kind is unavailable during cluster startup just because the cluster is forming and the other nodes have not yet registered completely.
allowing the caller to wait for a short period of time and then succeed, would be a better approach than forcing user to do retry